### PR TITLE
mgmt/mcumgr/lib: Add OS group command for obtaining mcumgr parameters

### PR DESCRIPTION
**Pull request by de-nordic**
The PR adds two commits that:
 1) add documentation for new OS management command that allows to obtain MCUMGR parameters;
 2) add implementation for that commit together with Kconfig options.

Due to lack of better suggestions to https://github.com/zephyrproject-rtos/zephyr/issues/42325 this is the best thing we can have now.

### CODE REVIEW

File: doc/services/device_mgmt/smp_groups/smp_group_0.rst [hunk 1]
```diff
@@ -456,3 +458,63 @@ where:
...
+    +========+==============+================+
+    | ``2``  | ``0``        |  ``6``         |
+    +--------+--------------+----------------+
+
+CBOR data of response:
+
+.. code-block:: none
+
+    {
+        (str)"buff_size"    : (uint)
```
Comment (philips77) [inline doc/services/device_mgmt/smp_groups/smp_group_0.rst:502]: Why buff, not buf?

File: doc/services/device_mgmt/smp_groups/smp_group_0.rst [hunk 2]
```diff
@@ -456,3 +458,63 @@ where:
     | "rc"                  | :ref:`mcumgr_smp_protocol_status_codes`;          |
     |                       | may not appear if 0                               |
     +-----------------------+---------------------------------------------------+
+
+MCUMGR Parameters
+*****************
+
+Used to obtain parameters of mcumgr library.
+
+System reset request
```
Comment (lairdjm) [inline doc/services/device_mgmt/smp_groups/smp_group_0.rst:467]: Don't think this title is correct

File: doc/services/device_mgmt/smp_groups/smp_group_0.rst [hunk 3]
```diff
@@ -456,3 +458,63 @@ where:
...
+    +========+==============+================+
+    | ``0``  | ``0``        |  ``6``         |
+    +--------+--------------+----------------+
+
+The command sends empty CBOR map as data.
+
+System reset response
+=====================
+
+System reset response header fields
```
Comment (lairdjm) [inline doc/services/device_mgmt/smp_groups/smp_group_0.rst:486]: Nor these

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: doc/services/device_mgmt/smp_groups/smp_group_0.rst (hunk 2 above)
Comment (de-nordic) [reply doc/services/device_mgmt/smp_groups/smp_group_0.rst:467]: Will fix.

File: doc/services/device_mgmt/smp_groups/smp_group_0.rst (hunk 3 above)
Comment (de-nordic) [reply doc/services/device_mgmt/smp_groups/smp_group_0.rst:486]: Will fix.

Comment (de-nordic) [reply doc/services/device_mgmt/smp_groups/smp_group_0.rst]: Done

File: doc/services/device_mgmt/smp_groups/smp_group_0.rst (hunk 2 above)
Comment (de-nordic) [reply doc/services/device_mgmt/smp_groups/smp_group_0.rst]: Done

File: doc/services/device_mgmt/smp_groups/smp_group_0.rst (hunk 1 above)
Comment (de-nordic) [reply doc/services/device_mgmt/smp_groups/smp_group_0.rst]: Done

**Comment by de-nordic** [comment]
Fixed, moved to zcbor, tested.

**Review by nordicjm — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**